### PR TITLE
Add :rst:role:`math:numref` role to refer equations (Same as :rst:role:`eq`)

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Feature

### Purpose
- Add `math:numref` role as an alias of `eq` role.

### CODE REVIEW

File: doc/usage/restructuredtext/domains.rst [hunk 1]
```diff
@@ -1325,6 +1325,25 @@ These roles are provided to refer to the described objects:
...
               rst:role
 
+.. _math-domain:
+
+The Math Domain
+---------------
+
+The math domain (name **math**) provides the following roles::
+
+.. rst:role:: rst:numref
```
Comment (jfbu) [inline doc/usage/restructuredtext/domains.rst]: ``math:numref`` ?

File: doc/usage/restructuredtext/domains.rst [hunk 2]
```diff
@@ -1325,6 +1325,25 @@ These roles are provided to refer to the described objects:
...
+
+   Role for cross-referencing equations defined by :rst:dir:`math` directive
+   via their label.  Example::
+
+      .. math:: e^{i\pi} + 1 = 0
+         :label: euler
+
+      Euler's identity, equation :eq:`euler`, was elected one of the most
```
Comment (jfbu) [inline doc/usage/restructuredtext/domains.rst]: did you mean to use ``:math:numref:`euler` `` here ?

**Review by jfbu — COMMENTED** [review]
LGTM with nits

**Comment by jfbu** [comment]
are you planning to deprecate `:eq:` in future?

File: doc/usage/restructuredtext/domains.rst (hunk 1 above)
Comment (tk0miya) [reply doc/usage/restructuredtext/domains.rst]: oh... thanks. fixed

File: doc/usage/restructuredtext/domains.rst (hunk 2 above)
Comment (tk0miya) [reply doc/usage/restructuredtext/domains.rst]: right. done

**Comment by tk0miya** [comment]
No, I'd like to keep `:eq:` role for a while (or forever?).
I think there are many document contain the old named role notations. And I believe documents have very longer life than program.

**Comment by jfbu** [comment]
> I think there are many document contain the old named role notations. And I believe documents have very longer life than program.

agreed :) I am annoyed that LaTeX babel-french for example will one day remove legacy "frenchb" in favour of only "french" and rename "\frenchbsetup" to "\frenchsetup". This means old sources from ten years ago may need manual edit to compile... bad. (my example here is for direct use of LaTeX, not via Sphinx...)

**edit** another better example is the brutal suppression by KOMA-script classes of support of `\bf`, `\rm` etc... about one year ago after more than 20 years of support. This is really silly decision, because the macros are still defined in LaTeX standard classes. It is impossible to understand why one day the authors decide to stop supporting them, breaking many documents which use LaTeX packages which refer to it and are perhaps now unmaintained.

**Review by jfbu — APPROVED** [review]

**Comment by tk0miya** [comment]
I can't promise all directives and roles will keep supported in 10+ yrs later. But we have to care to keep them as possible.
On the other hand, sustainability is also important for all of maintainers. So we have to discuss how to keep readability and maintainability of our code.

Anyway,  I think keeping `:eq:` role is not hard at all to us :-)

**Comment by tk0miya** [comment]
Merged. Thank you for reviewing.

**Pull request merged**